src/preferences/dialog/dlgprefcolors.cpp

```diff
@@ -77,6 +84,7 @@ DlgPrefColors::~DlgPrefColors() {
 void DlgPrefColors::slotUpdate() {
     comboBoxHotcueColors->clear();
     comboBoxTrackColors->clear();
+    bKeyColorsEnabled->setChecked(m_bKeyColorsEnabled);
```

Comment:
slotUpdate() is supposed to read the config values and update the view accordingly.
Here, the previous (GUI) value is used (that's why the box is still unchecked even though the setting is ON), and it's not obvious that `m_colorPaletteSettings.get..` queries the config.
Please read the config with `m_pConfig->getValue([color setting])`.